### DISCUSSION TITLE
Safari Light, Safari Dark themes

### DISCUSSION THREAD

**Discussion by vladstudio** [mikekazakov/nimble-commander#107] [Show and tell]
I created two themes that, I think, play nicer with the default MacOS look (I borrowed the colours from Safari with the default accent color).

[Safari Dark.json](https://github.com/mikekazakov/nimble-commander/files/14072353/Safari.Dark.json)

[Safari Light.json](https://github.com/mikekazakov/nimble-commander/files/14072354/Safari.Light.json)

<img width="1136" alt="image" src="https://github.com/mikekazakov/nimble-commander/assets/914320/22b6ec8f-06d3-4867-ba62-1cbad8c23531">

<img width="1136" alt="image" src="https://github.com/mikekazakov/nimble-commander/assets/914320/9a19efaf-3d72-495a-b19a-398fb7921b38">

**Comment by mikekazakov** [comment]
Hey, that looks cool!
What changes did you have to make, comparing to the prebuilt Light and Dark themes?
Would you dare to make a PR that includes them into a default set of themes in NC? :)

**Reply by vladstudio** [reply]
Thanks!

> What changes did you have to make

hmhm well I changed a bunch of colours :-)

> Would you dare to make a PR

I can try!

**Reply by vladstudio** [reply]
@mikekazakov - I cloned the repo and opened it in VSCode but failed to find where the themes are stored. Am I required to use XCode? (Don't use it). If so, I'd ask you to make a PR, never made it either :-)

**Reply by mikekazakov** [reply]
Default themes are stored in `Source/NimbleCommander/NimbleCommander/Resources/Config.json`: https://github.com/mikekazakov/nimble-commander/blob/73b4daa13ae8cf76474b4f7237a8c95457543a0d/Source/NimbleCommander/NimbleCommander/Resources/Config.json#L511

I'm pretty sure that nowadays you can fork the repository in GitHub and edit the file directly in the browser without even cloning locally, but that's a matter of preferences of course. Once you commit your changes to the forked repository GitHub will automatically build a new version of NC out of it, the dmg file will be downloadable from the Artifacts section of each integration.

**Reply by vladstudio** [reply]
whew, I think I created a PR!

**Comment by mikekazakov** [comment]
I guess this discussion can be closed - thanks for bringing more options to Nimble Commander's appearance!

**Comment by mikekazakov** [comment]
I'm using the new themes now and noticed that the Dark variant has a white background in the viewer:
`"viewerBackgroundColor": "#FFFFFF"` - that seems odd. Could you take a look at this color?

**Reply by vladstudio** [reply]
Oops. Fixed with `"viewerBackgroundColor": "#202020"`

**Reply by mikekazakov** [reply]
Great, any PR for that? :)

**Reply by vladstudio** [reply]
I hope I did it right.

**Reply by mikekazakov** [reply]
Ah, ok, I pulled that commit into main, thanks!

**Comment by mikekazakov** [comment]
@vladstudio Themes recently got colours for syntax highlighting, would you like to tune these for Safari Light/Dark?
I've added the same colours as in Light/Dark, but I'm pretty sure they look quite odd in the Safari themes.

**Comment by vladstudio** [comment]
Hey! I just downloaded the latest version and syntax highlighting does not seem to work for me. All file types use single shade of grey text. Did I miss anything?

https://github.com/mikekazakov/nimble-commander/assets/914320/671ee842-7f5d-4c81-b8b3-6f0746a790b5

**Comment by mikekazakov** [comment]
I'm gradually adding syntax settings for different languages, so far Bash, C++, JSON and YAML are supported:
https://github.com/mikekazakov/nimble-commander/blob/main/Source/NimbleCommander/NimbleCommander/Resources/SyntaxHighlighting/Main.json
The JSON file should have been highlighted.
Perhaps the build you picked didn't have the color definitions for Safari Light (i.e. it was a previous build) or maybe NC thought that you have a custom version of this theme and didn't load the settings from the version that comes with the app.
Alternatively you can check for errors via Alt+Help->Debug->Show Logs->Viewer->Trace.

**Reply by mikekazakov** [reply]
Oh, just to clarify, this functionality is available only in nightly builds, not in an "official" release yet.

**Reply by vladstudio** [reply]
where are nightly builds?

**Reply by mikekazakov** [reply]
Actions->Nightly Build: https://github.com/mikekazakov/nimble-commander/actions/workflows/nightly.yml
Go to the latest one and click on the "nimble-commander-nightly" artefact at the bottom of the page.
